source/MaterialXFormat/XmlIo.cpp

```diff
@@ -106,6 +107,11 @@ void elementToXml(ConstElementPtr elem, xml_node& xmlNode, const XmlWriteOptions
             {
                 if (!writtenSourceFiles.count(sourceUri))
                 {
+                    // Check to see if we have flagged this XInclude to be ignored, if so skip it.
+                    if (std::find(ignoredXIncludes.begin(), ignoredXIncludes.end(), sourceUri) != ignoredXIncludes.end())
```

Comment:
Searches for xinclude in list of ignored xincludes, if found, skips writing out this xinclude.